`sto_iter` compile fails on `math_chebyshev` on clang

I'm compiling abacus with `clang` compiler, and the error generates as follow:
``` yaml
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `void ModuleBase::Chebyshev<double>::calcoef_real<Sto_Func<double> >(Sto_Func<double>*, double (Sto_Func<double>::*)(double))':
[build] /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:20: undefined reference to `Sto_Func<double>::nfd(double)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:44: undefined reference to `Sto_Func<double>::nfd(double)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `Stochastic_Iter::calne()':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:286: undefined reference to `Sto_Func<double>::fd(double)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `Stochastic_Iter':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:10: undefined reference to `Sto_Func<double>::Sto_Func()'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `~Stochastic_Iter':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:20: undefined reference to `ModuleBase::Chebyshev<double>::~Chebyshev()'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `Stochastic_Iter::init(int, int*)':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:26: undefined reference to `ModuleBase::Chebyshev<double>::Chebyshev(int)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `Stochastic_Iter::itermu(int&)':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:243: undefined reference to `Sto_Func<double>::fd(double)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `void ModuleBase::Chebyshev<double>::tracepolyA<Stochastic_hchi>(Stochastic_hchi*, void (Stochastic_hchi::*)(std::complex<double>*, std::complex<double>*, int), std::complex<double>*, int, int, int)':
[build] /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:365: undefined reference to `ModuleBase::Chebyshev<double>::ddot_real(std::complex<double> const*, std::complex<double> const*, int, int, int)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:366: undefined reference to `ModuleBase::Chebyshev<double>::ddot_real(std::complex<double> const*, std::complex<double> const*, int, int, int)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:372: undefined reference to `ModuleBase::Chebyshev<double>::ddot_real(std::complex<double> const*, std::complex<double> const*, int, int, int)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `void ModuleBase::Chebyshev<double>::calcoef_real<Sto_Func<double> >(Sto_Func<double>*, double (Sto_Func<double>::*)(double))':
[build] /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:20: undefined reference to `Sto_Func<double>::nfdlnfd(double)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:44: undefined reference to `Sto_Func<double>::nfdlnfd(double)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `Stochastic_Iter::sum_stoband(Stochastic_WF&)':
[build] /home/cc/abacus-develop/source/src_pw/sto_iter.cpp:322: undefined reference to `Sto_Func<double>::fdlnfd(double)'
[build] /usr/bin/ld: source/src_pw/CMakeFiles/pw.dir/sto_iter.cpp.o: in function `void ModuleBase::Chebyshev<double>::calcoef_real<Sto_Func<double> >(Sto_Func<double>*, double (Sto_Func<double>::*)(double))':
[build] /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:20: undefined reference to `Sto_Func<double>::nxfd(double)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:44: undefined reference to `Sto_Func<double>::nxfd(double)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:20: undefined reference to `Sto_Func<double>::nroot_fd(double)'
[build] /usr/bin/ld: /home/cc/abacus-develop/source/src_pw/../module_base/math_chebyshev_def.h:44: undefined reference to `Sto_Func<double>::nroot_fd(double)'
[build] clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I suggest using `template<>` for [template specialization](https://en.cppreference.com/w/cpp/language/template_specialization) at `__MIX_PRECISION`, and in advance move template class definitions to header file. 
However, this issue might be caused by the difference on behavior between gcc and clang. ([ref](https://stackoverflow.com/questions/52725022/gcc-clang-disagree-on-partial-specialization-of-template-template-parameter))

--- a/source/src_pw/sto_func.cpp
+++ b/source/src_pw/sto_func.cpp
@@ -1,10 +1,5 @@
 #include "sto_func.h"
 #include "occupy.h"
-//we only have two examples: REAL and float.
-template class Sto_Func<double>;
-#ifdef __MIX_PRECISION
-template class Sto_Func<float>;
-#endif
 
 template<typename REAL>
 Sto_Func<REAL>::Sto_Func()
@@ -80,7 +75,7 @@ REAL Sto_Func<REAL>:: fdlnfd(REAL e)
     else
     {
         REAL f = 1 / (1 + exp(e_mu));
-        return (f * log(f) + (1.0-f) * log(1.0-f)); 
+        return (f * log(f) + (1.0-f) * log(1.0-f));
     }
 }
 
@@ -97,7 +92,12 @@ REAL Sto_Func<REAL>:: nfdlnfd(REAL e)
     else
     {
         REAL f = 1 / (1 + exp(ne_mu));
-        return f * log(f) + (1-f) * log(1-f); 
+        return f * log(f) + (1-f) * log(1-f);
     }
 }
 
+//we only have two examples: REAL and float.
+template class Sto_Func<double>;
+#ifdef __MIX_PRECISION
+template class Sto_Func<float>;
+#endif

